platform/platform-configuration-listener/src/main/java/org/codice/ddf/configuration/store/SystemConfigurationMigrator.java

```diff
@@ -0,0 +1,193 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SystemConfigurationMigrator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SystemConfigurationMigrator.class);
+
+    private static final String KEYSTORE_SYSTEM_PROP = "javax.net.ssl.keyStore";
+
+    private static final String TRUSTSTORE_SYSTEM_PROP = "javax.net.ssl.trustStore";
+
+    private static final String FILE_CONTAINING_CRL_LOCATION = "etc/ws-security/server/encryption.properties";
+
+    private static final String WS_SECURITY_DIR = "etc/ws-security";
+
+    private static final String SECURITY_DIRECTORY = "security";
+
+    private static final String PDP_POLICIES_DIR = "etc/pdp";
+
+    private static final String CRL_PROP_KEY = "org.apache.ws.security.crypto.merlin.x509crl.file";
+
+    private static final String SYSTEM_PROPERTIES = "etc/system.properties";
+
+    private static final String USERS_PROPERTIES = "etc/users.properties";
+
+    private final Path ddfHome;
+
+    public SystemConfigurationMigrator(Path ddfHome) {
+        this.ddfHome = ddfHome;
+    }
+
+    public void export(Path exportDirectory) throws MigrationException {
+        exportSecurity(exportDirectory);
+        exportSystemFiles(exportDirectory);
+    }
+
+    private void exportSecurity(Path exportDirectory) throws MigrationException {
+        exportSecurityDirectory(exportDirectory);
+        exportKeystores(exportDirectory);
+        exportWsSecurity(exportDirectory);
+        exportCrl(exportDirectory);
+        exportPdpPolicies(exportDirectory);
+    }
+
+    private void exportSystemFiles(Path exportDirectory) throws MigrationException {
+        exportSystemProperties(exportDirectory);
+        exportUsersProperties(exportDirectory);
+    }
+
+    private void exportKeystores(Path exportDirectory) throws MigrationException {
+        exportKeystore(exportDirectory);
+        exportTruststore(exportDirectory);
+    }
+
+    private void exportKeystore(Path exportDirectory) throws MigrationException {
+        String keystore = getProperty(KEYSTORE_SYSTEM_PROP);
+        if (StringUtils.isBlank(keystore)) {
+            throw new MigrationException(String.format(
+                    "Cannot find keystore. System property %s is not set.", KEYSTORE_SYSTEM_PROP));
+        }
+        Path source = ddfHome.resolve(keystore);
+        Path destination = constructDestination(source, exportDirectory);
+        copyFile(source, destination);
+    }
+
+    private void exportTruststore(Path exportDirectory) throws MigrationException {
+        String truststore = getProperty(TRUSTSTORE_SYSTEM_PROP);
+        if (StringUtils.isBlank(truststore)) {
```

Comment:
Do you think we could move the `isBlank()` check inside the `getProperty()` method to avoid the duplication?